lib/train/transports/azure.rb

```diff
@@ -54,14 +56,21 @@ def platform
         force_platform!('azure', @platform_details)
       end
 
+      def graph_client
```

Comment:
This may be confusing as the main client object for all azure clients is under this. I think it may be clearer to just put some logic into the `azure_client` method to pick out the graph client. This way we don't end up with `inspec.backend.graph_client` in inspec, which is a type of `azure_client`.

Reply:
I did initially have have this logic within the `azure_client` method, however @dmccown and I thought this convenience method was clearer. We do indeed then have `inspec.backend.graph_client` present in Inspec [here](https://github.com/inspec/inspec-azure/blob/dmccown/replace-connection-with-train/libraries/azurerm_resource.rb#L29) - is this bad?

Reply:
After talking this through, we'll change `azure_client` to handle the case that a client passes in `::Azure::GraphRbac::Profiles::Latest::Client`. The returned client will be fully hydrated and ready for consumption. So essentially it will fix an existing bug in cases you asked for a Graph client and the returned client didn't work.